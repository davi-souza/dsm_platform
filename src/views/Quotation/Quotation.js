import React, {useState} from 'react';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import QuotationItem from '../../components/QuotationItem';
import QuotationBasket from '../../components/QuotationBasket';
import QuotationItemsContext from '../../contexts/quotationItemsContext';
import './Quotation.scss';

const mockItems = [
	{
		name: 'Turbina do motor',
		qtd: 1,
		material: 'Iron',
		unitPrice: {
			amount: 10000,
			currency: 'R$',
		},
	},
	{
		name: 'Corpo do tubo',
		qtd: 1,
		material: "Steel",
		unitPrice: {
			amount: 20000,
			currency: 'R$',
		},
	},
];

function Quotation() {
	const [items, setItems] = useState(mockItems);

	function setItemQtd(itemIndex, newQtd) {
		/**	This function updates the qtd of a item
		 * Params:
		 * - itemIndex: index of the item in the array
		 * - newQtd: new qtd value
		 */
		if (newQtd < 0) {
			return;
		}

		const newItems = [...items];

		newItems[itemIndex].qtd = newQtd;

		setItems(newItems);
	}

	return (
		<Container maxWidth="lg">
			<QuotationItemsContext.Provider 
				value={{
					items,
					setItemQtd,
				}}
			>
				<Grid container spacing={1}>
					<Grid item xs={12}>
						<Button>Adicionar pe√ßa</Button>
					</Grid>
					<Grid
						item
						xs={12}
						md={9}
					>
						{items.map((item, index) => (
							<QuotationItem
								key={'quotation-item-' + index}
								{...Object.assign(
									item,
									{index},
									{setItemQtd}
								)}
							/>
						))}
					</Grid>
					<Grid item xs={12} md={3}>
						<QuotationBasket />
					</Grid>
				</Grid>
			</QuotationItemsContext.Provider>
		</Container>
	);
}

export default Quotation;
