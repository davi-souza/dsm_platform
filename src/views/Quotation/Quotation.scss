.quotation-view {
	&__container {
		min-height: calc(100% - 5rem);
	}
}
