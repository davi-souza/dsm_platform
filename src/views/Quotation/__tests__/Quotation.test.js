import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Quotation from '../Quotation';

configure({ adapter: new Adapter() });

describe('Quotation tests', () => {
	it('should render without crash', () => {
		const component = shallow(<Quotation />);
		expect(component).toMatchSnapshot();
	});
});
