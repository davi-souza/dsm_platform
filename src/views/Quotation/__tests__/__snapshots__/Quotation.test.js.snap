// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotation tests should render without crash 1`] = `ShallowWrapper {}`;
